CMakeLists.txt

```diff
@@ -236,9 +236,12 @@ function(install_windows_deps)
   set(ENV{ARCH} ${ARCH})
   execute_process(
     COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/tools/setup-buildtools.cmd)
-  set(CMAKE_TOOLCHAIN_FILE
-      ${CMAKE_CURRENT_SOURCE_DIR}/tools/vcpkg/scripts/buildsystems/vcpkg.cmake
-      PARENT_SCOPE)
+  # set(CMAKE_TOOLCHAIN_FILE
+  #     ${CMAKE_CURRENT_SOURCE_DIR}/tools/vcpkg/scripts/buildsystems/vcpkg.cmake
+  #     PARENT_SCOPE)
+  message("Make sure that vcpkg.cmake is set as the CMAKE_TOOLCHAIN_FILE at the START of the cmake build process! 
```

Comment:
Can we print this message only when we do not set toolchain file?

```cmake
if(NOT CMAKE_TOOLCHAIN_FILE)
  message(WARNING "Make sure that vcpkg.cmake is set as the CMAKE_TOOLCHAIN_FILE at the START of the cmake build process!
Can be command-line arg (cmake -DCMAKE_TOOLCHAIN_FILE=...) or set in your editor of choice.")
endif()
```

Reply:
The issue is the cmakefile tries to set the CMAKE_TOOLCHAIN_FILE in the cache at the start. This does not work unless its at the very start of the build process.

![image](https://user-images.githubusercontent.com/39804642/216082437-7af26595-5555-478a-b883-f3ffe4261151.png)

So its set, but it wasn't detecting my vcpkg unless I set it manually using `cmake -DCMAKE_TOOLCHAIN_FILE=... `
It may have to do with the submodules thinking the source directory is different, but gtest was not getting detected even after running install_windows_deps()

Reply:
If we use `cmake -DCMAKE_TOOLCHAIN_FILE=...` to set toolchain, we should also replace `NOT DEFINED CMAKE_TOOLCHAIN_FILE` by (NOT DEFINED CMAKE_TOOLCHAIN_FILE AND NOT DEFINED CACHE{CMAKE_TOOLCHAIN_FILE}).

Reply:
It can also just be removed entirely since it doesn't work

Reply:
Sorry for my mistake, `DEFINED <VAR NAME>` can also be used to check cached variables.
LGTM after format problem is solved.

Reply:
@bachittle I am still not clear how CMakeCache.txt affects the validity of `CMAKE_TOOLCHAIN_FILE` here. Could you please elaborate a little more? Thanks.
> The issue is the cmakefile tries to set the CMAKE_TOOLCHAIN_FILE in the cache at the start. This does not work unless its at the very start of the build process.

Reply:
so I've been doing some digging and I think the issue is that you have to set the CMAKE_TOOLCHAIN_FILE variable very early on in the build process as it indicates to cmake that other top level includes need to be set for things to work properly. When this is set later on it can't find gtest, so maybe some hidden higher-level include is being skipped. Either that or I am doing something incorrectly...

Reply:
Wondering what is your full CMake command line.

Reply:
Asking because I'd like to check if OTLP is not enabled in your case. For the `WITH_OTLP=ON` case, `${CMAKE_TOOLCHAIN_FILE}` is included explicitly in below line before run `find_package(...)` again, so there is no need to set the variable `CMAKE_TOOLCHAIN_FILE` at very early time. Probably we should do the same for the call to `find_package(GTest)` which seems the issue here.

https://github.com/open-telemetry/opentelemetry-cpp/blob/c4c8ee053f68d052f2287dd15dffcf42cc9e98a8/CMakeLists.txt#L321

Reply:
@bachittle ping on my comment here.

Reply:
sorry I didn't see this, will take a look

Reply:
![image](https://user-images.githubusercontent.com/39804642/221689448-b77a1721-eaa3-43d0-a569-ee9c8d5c35a9.png)
these are my cmake flags that are set using cmake-gui. notice the WITH_OTLP is not set, so I am just building the tests.

Reply:
Thanks @bachittle. This does confirm that OTLP is not enabled. Instead of commenting out the set of `CMAKE_TOOLCHAIN_FILE`, can we add a line like below in the call after `find_package(GTest REQUIRED)` in the same file?

https://github.com/open-telemetry/opentelemetry-cpp/blob/504644380a9482258604ad644531c643937b216c/CMakeLists.txt#L331

Reply:
I tried this and it still failed, check latest commit: https://github.com/open-telemetry/opentelemetry-cpp/pull/1887/commits/2bd872bc1cbdab770936c61a8fbc492d65b89575. Let me know if I should try putting this somewhere else.